### PR TITLE
Provide an option to disable tests and examples

### LINKED ISSUE #30: Make tests and examples optional
People installing arborx should not need to compile them. In addition, it would reduce the list of dependencies, as we won't need `boost`.

### PR DESCRIPTION
**Pull request by aprokop**
Fix #30

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,8 +16,12 @@ if(KOKKOS_DEVICES MATCHES "Cuda")
   target_link_directories(ArborX INTERFACE ${CUDA_LIBRARY_DIR})
 endif()
 
-enable_testing()
-
 add_subdirectory(src)
-add_subdirectory(test)
-add_subdirectory(examples)
+
+option(ArborX_ENABLE_TESTS "Enable tests and examples" ON)
```
Comment (dalg24) [inline CMakeLists.txt]: Please define two separate options, one for the tests and one for the examples

**Comment by aprokop** [comment]
I thought about it, but could not figure out a use case for when we only want one.

**Comment by aprokop** [comment]
After a discussion, decided to split the options.

**Comment by aprokop** [comment]
Separated options for examples and tests.

**Pull request merged**